### PR TITLE
luci-app-dcwapd: Add Dual Channel Wi-Fi AP Daemon Management LuCI pages

### PR DESCRIPTION
**Pull request by csonsino**
Add web interface configuration pages for the Dual Channel Wi-Fi AP Daemon.

Signed-off-by: Carey Sonsino <careys@edgewaterwireless.com>

### CODE REVIEW

File: applications/luci-app-dcwapd/Makefile [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+endef
+
+define Build/Compile
+endef
+
+define Package/luci-app-dcwapd/install
+  $(CP) ./files/* $(1)
+endef
+
+$(eval $(call BuildPackage,luci-app-dcwapd))
```
Comment (feckert) [inline applications/luci-app-dcwapd/Makefile]: Please have a look on other Makefile in this application folder. Use this as a template.

File: applications/luci-app-dcwapd/files/usr/lib/lua/luci/model/cbi/dcwapd/dcwapd.lua [hunk 2]
```diff
@@ -0,0 +1,237 @@
+-- Copyright 2018 EWSI
+-- Licensed to the public under the Apache License 2.0.
+
+local wa = require "luci.tools.webadmin"
+local fs = require "nixio.fs"
+local ntm = require "luci.model.network"
+local uc = require "luci.model.uci".cursor()
+
+require "luci.sys"
```
Comment (feckert) [inline applications/luci-app-dcwapd/files/usr/lib/lua/luci/model/cbi/dcwapd/dcwapd.lua]: Please make this variable local

> local sys = require "luci.sys"

File: applications/luci-app-dcwapd/files/usr/lib/lua/luci/model/cbi/dcwapd/dcwapd.lua [hunk 3]
```diff
@@ -0,0 +1,237 @@
...
+proto.rmempty = false
+proto.default = translate("*")
+
+-- Destination Port
+dstport = s:option(Value, "dest_port", translate("Destination port"))
+dstport.rmempty = false
+dstport:value("*", translate("*"))
+dstport.default = translate("*")
+
+return m
```
Comment (feckert) [inline applications/luci-app-dcwapd/files/usr/lib/lua/luci/model/cbi/dcwapd/dcwapd.lua]: Please remove the lines on config option that are comments
And also please make the variables for the uci values local on the top of this file
`local dstport, proto,`  and so on

File: applications/luci-app-dcwapd/files/usr/lib/lua/luci/model/cbi/dcwapd/dcwapd.lua [hunk 4]
```diff
@@ -0,0 +1,237 @@
...
+proto:value("udp", "UDP")
+proto:value("icmp", "ICMP")
+proto.rmempty = false
+proto.default = translate("*")
+
+-- Destination Port
+dstport = s:option(Value, "dest_port", translate("Destination port"))
+dstport.rmempty = false
+dstport:value("*", translate("*"))
+dstport.default = translate("*")
```
Comment (feckert) [inline applications/luci-app-dcwapd/files/usr/lib/lua/luci/model/cbi/dcwapd/dcwapd.lua]: Do we realy need to translate `*` ?

Comment (csonsino) [reply applications/luci-app-dcwapd/files/usr/lib/lua/luci/model/cbi/dcwapd/dcwapd.lua]: No, I suppose that we do not.

**Comment by csonsino** [comment]
The [PR](https://github.com/openwrt/packages/pull/9972) on which this was dependent has been merged into master.

File: applications/luci-app-dcwapd/luasrc/model/cbi/dcwapd/dcwapd.lua [hunk 5]
```diff
@@ -0,0 +1,223 @@
...
+-- Licensed to the public under the Apache License 2.0.
+
+local wa = require "luci.tools.webadmin"
+local fs = require "nixio.fs"
+local ntm = require "luci.model.network"
+local uc = require "luci.model.uci".cursor()
+
+local sys = require "luci.sys"
+
+-- require "luci.debug".trap_memtrace("l", "/tmp/trace")
```
Comment (feckert) [inline applications/luci-app-dcwapd/luasrc/model/cbi/dcwapd/dcwapd.lua]: Please remove this comment this is not needed

File: applications/luci-app-dcwapd/luasrc/model/cbi/dcwapd/dcwapd.lua [hunk 6]
```diff
@@ -0,0 +1,223 @@
...
+uc:foreach("dcwapd", "filter",
+    function(s)
+        if s['.name'] then
+            filters:value(s['.name'])
+            filters.size = filters.size + 1
+        end
+    end)
+
+-- Filters section
+s = m:section(TypedSection, "filter", "Filters", "Filter Rules")
```
Comment (feckert) [inline applications/luci-app-dcwapd/luasrc/model/cbi/dcwapd/dcwapd.lua]: "Filter Rules" translation is missing

File: applications/luci-app-dcwapd/luasrc/model/cbi/dcwapd/dcwapd.lua [hunk 7]
```diff
@@ -0,0 +1,223 @@
...
+    ntm.init()
+    local net = ntm:get_wifinet(v)
+    if net then
+        ifaces:value(v)
+        ifaces.size = ifaces.size + 1
+    end
+end
+
+-- Filter sets section
+s = m:section(TypedSection, "filter-set", "Filter Sets", "Filter Sets")
```
Comment (feckert) [inline applications/luci-app-dcwapd/luasrc/model/cbi/dcwapd/dcwapd.lua]: "Filter Sets" translation is missing

File: applications/luci-app-dcwapd/luasrc/model/cbi/dcwapd/dcwapd.lua [hunk 8]
```diff
@@ -0,0 +1,223 @@
...
+s.anonymous = true
+
+-- Enabled state option
+enable = s:option(Flag, "enabled", translate("Enable"))
+enable.default = false
+enable.optional = false
+enable.rmempty = false
+
+-- Temp dir option
+tmpdir = s:option(Value, "tmpdir", "Temp Directory", "Temporary Directory for dcwapd file storage")
```
Comment (feckert) [inline applications/luci-app-dcwapd/luasrc/model/cbi/dcwapd/dcwapd.lua]: Translation

File: applications/luci-app-dcwapd/luasrc/model/cbi/dcwapd/dcwapd.lua [hunk 9]
```diff
@@ -0,0 +1,223 @@
...
+tmpdir.optional = false
+tmpdir.rmempty = false
+
+-- Channel sets section
+s = m:section(TypedSection, "channel-set", "Channel Sets", "")
```
Comment (feckert) [inline applications/luci-app-dcwapd/luasrc/model/cbi/dcwapd/dcwapd.lua]: Translation missing as well

File: applications/luci-app-dcwapd/luasrc/model/cbi/dcwapd/dcwapd.lua [hunk 10]
```diff
@@ -0,0 +1,223 @@
...
+local mac, filters
+local packetsize, srcip, srcport, proto, dstport
+
+m = Map("dcwapd", translate("Dual Channel Wi-Fi AP Daemon"),
+	translate("With <abbr title=\"Dual Channel Wi-Fi AP Daemon\">Dual Channel WiFi</abbr> you " ..
+		"can use two simultaneous Wi-Fi connections to decrease wireless traffic " ..
+		"congestion and increase throughput."))
+
+-- General section
+s = m:section(NamedSection, "general", "General", "General Settings")
```
Comment (feckert) [inline applications/luci-app-dcwapd/luasrc/model/cbi/dcwapd/dcwapd.lua]: Translation

**Comment by csonsino** [comment]
I think that all outstanding comments have been addressed, but let me know if I missed anything or if you see any other problems.  Thanks!

File: applications/luci-app-dcwapd/luasrc/controller/dcwapd.lua [hunk 11]
```diff
@@ -0,0 +1,15 @@
...
+
+module("luci.controller.dcwapd", package.seeall)
+
+function index()
+	if not nixio.fs.access("/etc/config/dcwapd") then
+		return
+	end
+	
+	local page
+
```
Comment (feckert) [inline applications/luci-app-dcwapd/luasrc/controller/dcwapd.lua]: please remove this whitspace <tab>

File: applications/luci-app-dcwapd/luasrc/controller/dcwapd.lua [hunk 12]
```diff
@@ -0,0 +1,15 @@
+-- Copyright 2019 EWSI
+-- Licensed to the public under the Apache License 2.0.
+
+module("luci.controller.dcwapd", package.seeall)
+
+function index()
+	if not nixio.fs.access("/etc/config/dcwapd") then
+		return
+	end
+
```
Comment (feckert) [inline applications/luci-app-dcwapd/luasrc/controller/dcwapd.lua]: please remove this whitspace <tab>

**Comment by feckert** [comment]
Sorry but one small nick pick needs to get fixed. The whitspaces are tabs and not spaces! So please changes this sorry. The for loops an function callbacks in model/cbi/dcwapd/dswapd.lua nee also a rework.
If this is fixed I will merge this. Thanks

**Comment by feckert** [comment]
Thanks for your contribution.

**Pull request merged**